app/inputs/ajax_filter_input.rb

```diff
@@ -30,7 +30,13 @@ def input_html_options
 
   # rubocop:disable Style/RescueModifier
   def get_selected_value(display_name)
-    filter_class = method.to_s.chomp("_id").classify.constantize rescue @object.klass
+    association_name = method.to_s.chomp("_id")
+    association = @object.klass.reflect_on_association(association_name)
+    if association.options && association.options.keys.include?(:class_name)
```

Comment:
There is a bug here. It's possible that `association` if method does not end in `_id`. For example, I have a filter on my index page that is called just `id`. I use it with an ajax powered select box (from activesupport_addons) for quick searching.

The previous code was more robust because it used `rescue @object.klass`, so we should bring that back.